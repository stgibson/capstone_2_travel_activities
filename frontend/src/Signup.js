import React from "react";
import BaseForm from "./BaseForm";
import Footer from "./Footer";

/**
 * Component for displaying form to create an account
 * @param {Object} param0 
 * @returns JSX code for rendering signup page
 */
const Signup = ({ signup }) => {
  const inputs = [
    { name: "username", label: "Username:", type: "text", default: "" },
    { name: "password", label: "Password", type: "password", default: "" }
  ]

  return (
    <>
      <BaseForm
        title="Sign Up"
        inputs={ inputs }
        btnText="Sign Up"
        submitCallback={ signup }
      />
      <Footer />
    </>
  );
};

export default Signup;