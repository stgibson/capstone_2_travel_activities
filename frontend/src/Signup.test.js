import React from "react";
import { render } from "@testing-library/react";
import Signup from "./Signup";

const signup = jest.fn();

it("renders without crashing", () => {
  render(<Signup signup={ signup } />);
});

it("matches snapshot", () => {
  const { asFragment } = render(<Signup signup={ signup } />);

  expect(asFragment()).toMatchSnapshot();
});
