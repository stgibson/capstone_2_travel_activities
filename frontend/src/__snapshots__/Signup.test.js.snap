// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <h2>
    Sign Up
  </h2>
  <form
    class=""
  >
    <div
      class="form-group"
    >
      <label
        class="form-label"
        for="username"
      >
        Username:
      </label>
      <input
        class="form-control"
        id="username"
        type="text"
        value=""
      />
    </div>
    <div
      class="form-group"
    >
      <label
        class="form-label"
        for="password"
      >
        Password
      </label>
      <input
        class="form-control"
        id="password"
        type="password"
        value=""
      />
    </div>
  </form>
  <button
    class="btn btn-primary"
    type="button"
  >
    Sign Up
  </button>
</DocumentFragment>
`;
